packages/core/src/util/cache.js

```diff
@@ -1,16 +1,78 @@
+import { list, tableFromArrays, tableFromIPC, tableToIPC,  uint8, utf8 } from "@uwdata/flechette";
+
 const requestIdle = typeof requestIdleCallback !== 'undefined'
   ? requestIdleCallback
   : setTimeout;
 
+/**
+ * @typedef {import('@uwdata/flechette').Table | Promise<import('@uwdata/flechette').Table>} CacheEntry
+ * @typedef {Object} Cache
+ * @property {(key: string) => CacheEntry | undefined} get Retrieves a value from the cache.
+ * @property {(key: string, value: CacheEntry) => any} set Stores a value in the cache and returns it.
+ * @property {() => void} clear Clears all entries in the cache.
+ * @property {() => Uint8Array | null} export Exports the cache as an array of bytes in Arrow IPC binary format or null if the cache is empty.
+ * @property {(data: ArrayBuffer | Uint8Array | Uint8Array[]) => void} import Imports the cache from an array of bytes in Arrow IPC binary format.
+ */
+
+/**
+ * Converts a cache instance into a byte buffer.
+ * @param {[string, import('@uwdata/flechette').Table][]} kv An array of key-value pairs representing
+ * the cache contents.
+ * @returns {Uint8Array | null} A byte buffer representing the cache or null if the cache is empty.
+ */
+const keyValuesToIPC = (kv) => {
+  const cache_object = kv.reduce(
+    (acc, [key, value]) => {
+      const bytes = tableToIPC(value, { format: 'stream' });
+      acc.key.push(key);
+      acc.value.push(bytes);
+      return acc;
+    },
+    { key: [], value: [] }
+  );
+  const table = tableFromArrays(cache_object, {
+    types: {
+      key: utf8(),
+      value: list(uint8())
```

Comment:
Shouldn't this be a [`binary`](https://idl.uw.edu/flechette/api/data-types#binary) type instead?